xlators/storage/posix/src/posix-common.c

```diff
@@ -721,15 +721,11 @@ posix_init(xlator_t *this)
 
     ret = dict_get_str(this->options, "hostname", &_private->hostname);
     if (ret) {
-        _private->hostname = GF_CALLOC(256, sizeof(char), gf_common_mt_char);
+        _private->hostname = gf_strdup(gf_gethostname());
```

Comment:
there are 'GF_FREE()' of this hostname in code, which may need to be removed now.

Reply:
I think it (GF_FREE) is still required

Reply:
Good point.